src/shape/2d_primitives.js

```diff
@@ -444,14 +444,22 @@ define(function (require) {
   * every angle at ninety degrees. By default, the first two parameters set
   * the location of the upper-left corner, the third sets the width, and the
   * fourth sets the height. The way these parameters are interpreted, however,
-  * may be changed with the rectMode() function.
+  * may be changed with the rectMode() function. If provided, the fifth, sixth
+  * seventh and eighth parameters, if specified, determine corner radius for
+  * the top-right, top-left, lower-right and lower-left corners, respectively.
+  * An omitted corner radius parameter is set to the value of the previously
+  * specified radius value in the parameter list.
   *
   * @method rect
-  * @param  {Number} a x-coordinate of the rectangle
-  * @param  {Number} b y-coordinate of the rectangle
-  * @param  {Number} c width of the rectangle
-  * @param  {Number} d height of the rectangle
-  * @return {p5}       the p5 object
+  * @param  {Number} x  x-coordinate of the rectangle.
+  * @param  {Number} y  y-coordinate of the rectangle.
+  * @param  {Number} w  width of the rectangle.
+  * @param  {Number} h  height of the rectangle.
+  * @param  {Number} [tl] optional radius of top-left corner.
+  * @param  {Number} [tr] optional radius of top-right corner.
+  * @param  {Number} [br] optional radius of bottom-right corner.
+  * @param  {Number} [bl] optional radius of bottom-left corner.
+  * @return {p5}          the p5 object.
   * @example
```

Comment:
It'd be good to add some additional `@example` blocks showing off this functionality